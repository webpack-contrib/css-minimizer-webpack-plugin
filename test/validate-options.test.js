import CssMinimizerPlugin from "../src";

it("validation", () => {
  /* eslint-disable no-new */
  expect(() => {
    new CssMinimizerPlugin({ test: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: "foo" });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: ["foo", "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: [/foo/, "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ test: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ test: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ include: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: "foo" });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: ["foo", "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: [/foo/, "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ include: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ include: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ exclude: /foo/ });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: "foo" });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: [/foo/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: [/foo/, /bar/] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: ["foo", "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: [/foo/, "bar"] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ exclude: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ exclude: [true] });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ minimizerOptions: {} });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ minimizerOptions: [{}] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ minimizerOptions: null });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({
      minimizerOptions: { colormin: true },
    });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ parallel: true });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ parallel: false });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ parallel: 2 });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ parallel: "2" });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ parallel: {} });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ minify() {} });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ minify: [() => {}] });
  }).not.toThrow();

  expect(() => {
    new CssMinimizerPlugin({ minify: true });
  }).toThrowErrorMatchingSnapshot();

  expect(() => {
    new CssMinimizerPlugin({ unknown: true });
  }).toThrowErrorMatchingSnapshot();
  /* eslint-enable no-new */
});
