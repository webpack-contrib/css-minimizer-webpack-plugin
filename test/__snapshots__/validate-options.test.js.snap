// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#test
   Details:
    * options.test should be an array:
      [RegExp | non-empty string, ...]
    * options.test should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test should be an instance of RegExp.
       * options.test should be a non-empty string."
`;

exports[`validation 2`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.test should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#test
   Details:
    * options.test[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.test[0] should be an instance of RegExp.
       * options.test[0] should be a non-empty string."
`;

exports[`validation 3`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#include
   Details:
    * options.include should be an array:
      [RegExp | non-empty string, ...]
    * options.include should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include should be an instance of RegExp.
       * options.include should be a non-empty string."
`;

exports[`validation 4`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.include should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#include
   Details:
    * options.include[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.include[0] should be an instance of RegExp.
       * options.include[0] should be a non-empty string."
`;

exports[`validation 5`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#exclude
   Details:
    * options.exclude should be an array:
      [RegExp | non-empty string, ...]
    * options.exclude should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude should be an instance of RegExp.
       * options.exclude should be a non-empty string."
`;

exports[`validation 6`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.exclude should be one of these:
   [RegExp | non-empty string, ...] | RegExp | non-empty string
   -> Filtering rules.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#exclude
   Details:
    * options.exclude[0] should be one of these:
      RegExp | non-empty string
      -> Filtering rule as regex or string.
      Details:
       * options.exclude[0] should be an instance of RegExp.
       * options.exclude[0] should be a non-empty string."
`;

exports[`validation 7`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minimizerOptions should be one of these:
   object { … } | [object { … }, ...] (should not have fewer than 1 item)
   -> Options for \`cssMinimizerOptions\`.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#minimizeroptions
   Details:
    * options.minimizerOptions should be an object:
      object { … }
    * options.minimizerOptions should be an array:
      [object { … }, ...] (should not have fewer than 1 item)"
`;

exports[`validation 8`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.parallel should be one of these:
   boolean | integer
   -> Use multi-process parallel running to improve the build speed.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#parallel
   Details:
    * options.parallel should be a boolean.
    * options.parallel should be an integer."
`;

exports[`validation 9`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.parallel should be one of these:
   boolean | integer
   -> Use multi-process parallel running to improve the build speed.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#parallel
   Details:
    * options.parallel should be a boolean.
    * options.parallel should be an integer."
`;

exports[`validation 10`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options.minify should be one of these:
   function | [function, ...] (should not have fewer than 1 item)
   -> Allows you to override default minify function.
   -> Read more at https://github.com/webpack-contrib/css-minimizer-webpack-plugin/#minify
   Details:
    * options.minify should be an instance of function.
    * options.minify should be an array:
      [function, ...] (should not have fewer than 1 item)"
`;

exports[`validation 11`] = `
"Invalid options object. Css Minimizer Plugin has been initialized using an options object that does not match the API schema.
 - options has an unknown property 'unknown'. These properties are valid:
   object { test?, include?, exclude?, minimizerOptions?, parallel?, warningsFilter?, minify? }"
`;
